tests/no-striping.cpp

```diff
@@ -0,0 +1,232 @@
+
+    props.video[0].camera.settings.binning = 1;
+    props.video[0].camera.settings.pixel_type = SampleType_u16;
+    props.video[0].camera.settings.shape = { .x = frame_width,
+                                             .y = frame_height };
+    // we may drop frames with lower exposure
+    props.video[0].camera.settings.exposure_time_us = 2e5;
+    props.video[0].max_frame_count = expected_frames_per_chunk;
+
+    OK(acquire_configure(runtime, &props));
+
+    const auto next = [](VideoFrame* cur) -> VideoFrame* {
+        return (VideoFrame*)(((uint8_t*)cur) + cur->bytes_of_frame);
+    };
+
+    const auto consumed_bytes = [](const VideoFrame* const cur,
+                                   const VideoFrame* const end) -> size_t {
+        return (uint8_t*)end - (uint8_t*)cur;
+    };
+
+    struct clock clock;
+    static double time_limit_ms = 20000.0;
+    clock_init(&clock);
+    clock_shift_ms(&clock, time_limit_ms);
+    OK(acquire_start(runtime));
+    {
+        uint64_t nframes = 0;
+        VideoFrame *beg, *end, *cur;
+        do {
+            struct clock throttle;
+            clock_init(&throttle);
+            EXPECT(clock_cmp_now(&clock) < 0,
+                   "Timeout at %f ms",
+                   clock_toc_ms(&clock) + time_limit_ms);
+            OK(acquire_map_read(runtime, 0, &beg, &end));
+            for (cur = beg; cur < end; cur = next(cur)) {
+                LOG("stream %d counting frame w id %d", 0, cur->frame_id);
+                CHECK(cur->shape.dims.width ==
+                      props.video[0].camera.settings.shape.x);
+                CHECK(cur->shape.dims.height ==
+                      props.video[0].camera.settings.shape.y);
+                ++nframes;
+            }
+            {
+                uint32_t n = consumed_bytes(beg, end);
+                OK(acquire_unmap_read(runtime, 0, n));
+                if (n)
+                    LOG("stream %d consumed bytes %d", 0, n);
+            }
+            clock_sleep_ms(&throttle, 100.0f);
+
+            LOG(
+              "stream %d nframes %d time %f", 0, nframes, clock_toc_ms(&clock));
+        } while (DeviceState_Running == acquire_get_state(runtime) &&
+                 nframes < props.video[0].max_frame_count);
+
+        OK(acquire_map_read(runtime, 0, &beg, &end));
+        for (cur = beg; cur < end; cur = next(cur)) {
+            LOG("stream %d counting frame w id %d", 0, cur->frame_id);
+            CHECK(cur->shape.dims.width ==
+                  props.video[0].camera.settings.shape.x);
+            CHECK(cur->shape.dims.height ==
+                  props.video[0].camera.settings.shape.y);
+            ++nframes;
+        }
+        {
+            uint32_t n = consumed_bytes(beg, end);
+            OK(acquire_unmap_read(runtime, 0, n));
+            if (n)
+                LOG("stream %d consumed bytes %d", 0, n);
+        }
+
+        CHECK(nframes == props.video[0].max_frame_count);
+    }
```

Comment:
Do we need to check frames here? Maybe cut. Alternatively, could factor this block into a `check_frame_count()` function.

```suggestion
```